src/cmd/sanity_test/sanity_tests.go

```diff
@@ -323,91 +328,120 @@ func checkOnedriveRestoration(
-		folderPermission[itemID] = permissionsIn(ctx, client, driveID, itemID, folderPermission[itemID])
+		if driveItem.GetFolder() == nil && driveItem.GetPackage() == nil {
+			continue
+		}
+
+		// currently we don't restore blank folders.
+		// skip permission check for empty folders
+		if ptr.Val(driveItem.GetFolder().GetChildCount()) == 0 {
+			fmt.Println("skipped empty folder: ", itemName)
+			continue
+		}
+
+		permissionIn(ctx, client, driveID, itemID, itemName, folderPermission)
+		getOneDriveChildFolder(ctx, client, driveID, itemID, itemName, fileSizes, folderPermission, startTime)
+	}
+
+	getRestoreData(ctx, client, *drive.GetId(), restoreFolderID, restoreFile, restoreFolderPermission, startTime)
+
+	for checkFolderName, checkfolderPer := range folderPermission {
+		fmt.Printf("checking for folder: %s \n", checkFolderName)
+
+		if len(checkfolderPer) < 1 {
+			fmt.Printf("no permissions found for folder : %s.", checkFolderName)
+			continue
+		}
+
+		if len(restoreFolderPermission[checkFolderName]) < 1 {
+			fmt.Printf("permissions are not equal")
+			fmt.Println("Item:", checkFolderName)
+			fmt.Println("Permission found: ", checkfolderPer)
+			fmt.Println("blank permission found in restore.")
+
+			os.Exit(1)
+		}
+
+		for i, orginalFolderPer := range checkfolderPer {
+			if !(orginalFolderPer.entityID != restoreFolderPermission[checkFolderName][i].entityID) &&
+				!slices.Equal(orginalFolderPer.roles, restoreFolderPermission[checkFolderName][i].roles) {
+				fmt.Printf("permissions are not equal")
+				fmt.Printf("*  expected role: %+v \n", orginalFolderPer.roles)
+				fmt.Printf("*  actual:  %+v \n", restoreFolderPermission[checkFolderName][i].roles)
+				fmt.Printf("* entitiy ID expected: %+v \n", orginalFolderPer.entityID)
+				fmt.Printf("* entitiy ID actual:  %+v \n", restoreFolderPermission[checkFolderName][i].entityID)
+				fmt.Println("Item:", checkFolderName)
+				os.Exit(1)
+			}
+		}
 	}
 
-	checkFileData(ctx, client, driveID, restoreFolderID, fileSizes, folderPermission)
+	for fileName, fileSize := range fileSizes {
+		if fileSize != restoreFile[fileName] {
+			fmt.Println("File size does not match:")
+			fmt.Println("*  expected:", fileSize)
+			fmt.Println("*  actual:", restoreFile[fileName])
+			fmt.Println("Item:", fileName)
+			os.Exit(1)
+		}
+	}
 
 	fmt.Println("Success")
 }
 
-func checkFileData(
+func getOneDriveChildFolder(
 	ctx context.Context,
 	client *msgraphsdk.GraphServiceClient,
-	driveID,
-	restoreFolderID string,
+	driveID, itemID, parentName string,
 	fileSizes map[string]int64,
-	folderPermission map[string]map[string][]string,
+	folderPermission map[string][]permissionInfo,
+	startTime time.Time,
 ) {
-	restored, err := client.
-		DrivesById(driveID).
-		ItemsById(restoreFolderID).
-		Children().
-		Get(ctx, nil)
+	response, err := client.DrivesById(driveID).ItemsById(itemID).Children().Get(ctx, nil)
 	if err != nil {
-		fatal(ctx, "getting child folder", err)
+		fatal(ctx, "getting permission", err)
```

Comment:
Is this correct?  The query retrieves child items, not permissions.

Reply:
Make bad!